src/Promenade.Controllers/HomeController.cs

```diff
@@ -1,21 +1,362 @@
 ﻿using System;
-using Ocuda.Promenade.Controllers.Abstract;
+using System.Collections.Generic;
+using System.Net.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Logging;
+using System.Linq;
+using Ocuda.Promenade.Controllers.Abstract;
+using Ocuda.Promenade.Controllers.ViewModels;
+using Ocuda.Promenade.Models.Entities;
+using BranchLocator.Helpers;
+using BranchLocator.Models;
+using Newtonsoft.Json;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
 
 namespace Ocuda.Promenade.Controllers
 {
+    [Route("")]
+
     public class HomeController : BaseController
     {
+        public const int DaysInAWeek = 7;
+
+        private readonly IConfiguration _config;
         private readonly ILogger<HomeController> _logger;
-        public HomeController(ILogger<HomeController> logger)
+
+        public HomeController(IConfiguration config,
+            ILogger<HomeController> logger)
         {
+            _config = config ?? throw new ArgumentNullException(nameof(config));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
         }
 
+        [Route("")]
         public IActionResult Index()
         {
             return View();
         }
+        [HttpGet("[action]")]
+        [HttpGet("[action]/{address}")]
+        [HttpGet("[action]/{address}/{mobile}")]
+        public async Task<IActionResult> LocateZip(string address = null, int? mobile = null)
+        {
+            if (!string.IsNullOrWhiteSpace(address))
+            {
+                using (var client = new HttpClient())
+                {
+                    try
+                    {
+                        var apikey = _config["APIKey"];
+
+                        var response = await client.GetAsync($"https://maps.googleapis.com/maps/api/geocode/json?address={address}&key={apikey}");
+                        response.EnsureSuccessStatusCode();
+
+                        var stringResult = await response.Content.ReadAsStringAsync();
+                        dynamic jsonResult = JsonConvert.DeserializeObject(stringResult);
+
+                        if (jsonResult.results.Count > 0)
+                        {
+                            var result = jsonResult.results[0];
+                            double latitude = result.geometry.location.lat;
+                            double longitude = result.geometry.location.lng;
+
+                            Location viewModel = LibraryLookup(latitude, longitude);
```

Comment:
This method that is being called should be named `LookupLocation`, not `LibraryLookup`.